In [1]:
# 0. 모듈 불러오기

import pandas as pd
import numpy as np

In [2]:
# 1. 데이터 불러오기

train_data = pd.read_json('data/train.json', encoding='UTF-8', typ = 'frame')
song_meta = pd.read_json('data/song_meta.json', encoding='UTF-8', typ = 'frame')
genre_gn_all = pd.read_json('data/genre_gn_all.json',encoding='UTF-8', typ = 'series')
genre_gn_all = pd.DataFrame(genre_gn_all, columns = ['gnr_name']).reset_index().rename(columns = {'index' : 'gnr_code'})

### 데이터를 확인해보자!

In [7]:
train_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 115071 entries, 0 to 115070
Data columns (total 6 columns):
 #   Column        Non-Null Count   Dtype 
---  ------        --------------   ----- 
 0   tags          115071 non-null  object
 1   id            115071 non-null  int64 
 2   plylst_title  115071 non-null  object
 3   songs         115071 non-null  object
 4   like_cnt      115071 non-null  int64 
 5   updt_date     115071 non-null  object
dtypes: int64(2), object(4)
memory usage: 5.3+ MB


In [3]:
song_meta.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 707989 entries, 0 to 707988
Data columns (total 9 columns):
 #   Column                  Non-Null Count   Dtype 
---  ------                  --------------   ----- 
 0   song_gn_dtl_gnr_basket  707989 non-null  object
 1   issue_date              707989 non-null  int64 
 2   album_name              707985 non-null  object
 3   album_id                707989 non-null  int64 
 4   artist_id_basket        707989 non-null  object
 5   song_name               707989 non-null  object
 6   song_gn_gnr_basket      707989 non-null  object
 7   artist_name_basket      707989 non-null  object
 8   id                      707989 non-null  int64 
dtypes: int64(3), object(6)
memory usage: 48.6+ MB


In [8]:
genre_gn_all.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 254 entries, 0 to 253
Data columns (total 2 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0   gnr_code  254 non-null    object
 1   gnr_name  254 non-null    object
dtypes: object(2)
memory usage: 4.1+ KB


#### train_data 크기 : 115071 (약 11만)
#### song 크기 : 707989 (약 70만)
#### genre_gn_all : 254

In [6]:
train_data.head()

tags      id  \
0                                                [락]   61281   
1                                           [추억, 회상]   10532   
2                                          [까페, 잔잔한]   76951   
3  [연말, 눈오는날, 캐럴, 분위기, 따듯한, 크리스마스캐럴, 겨울노래, 크리스마스,...  147456   
4                                               [댄스]   27616   

            plylst_title                                              songs  \
0                여행같은 음악  [525514, 129701, 383374, 562083, 297861, 13954...   
1                요즘 너 말야  [432406, 675945, 497066, 120377, 389529, 24427...   
2  편하게, 잔잔하게 들을 수 있는 곡.-  [83116, 276692, 166267, 186301, 354465, 256598...   
3  크리스마스 분위기에 흠뻑 취하고 싶을때  [394031, 195524, 540149, 287984, 440773, 10033...   
4               추억의 노래 ㅋ  [159327, 553610, 5130, 645103, 294435, 100657,...   

   like_cnt                updt_date  
0        71  2013-12-19 18:36:19.000  
1         1  2014-12-02 16:19:42.000  
2        17  2017-08-28 07:09:34.000  
3        33  2019-12-05 15:15:18.000  
4         9  2011-10-25 13:54:56.000

In [11]:
song_meta['id'].describe()

count    707989.000000
mean     353994.000000
std      204378.964203
min           0.000000
25%      176997.000000
50%      353994.000000
75%      530991.000000
max      707988.000000
Name: id, dtype: float64

In [12]:
train_data['id'].describe()

count    115071.000000
mean      76728.259501
std       44313.134619
min           1.000000
25%       38277.500000
50%       76732.000000
75%      115061.500000
max      153428.000000
Name: id, dtype: float64

In [10]:
genre_gn_all

gnr_code gnr_name
0     GN0100      발라드
1     GN0101   세부장르전체
2     GN0102      '80
3     GN0103      '90
4     GN0104      '00
..       ...      ...
249   GN2900      뮤지컬
250   GN2901   세부장르전체
251   GN2902    국내뮤지컬
252   GN2903    국외뮤지컬
253   GN3000    크리스마스

[254 rows x 2 columns]